cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -548,8 +548,10 @@ class InputState {
   // true when the queue is empty and, when memo may have future entries (the case of a
   // positive tolerance), when the memo is empty.
   // used when checking whether RHS is up to date with LHS.
-  bool CurrentEmpty() const {
-    return memo_.no_future_ ? Empty() : memo_.times_.empty() && Empty();
+  // NOTE: The emptiness must be decided by an atomic all to Empty() in caller, due to the
```

Comment:
```suggestion
  // NOTE: The emptiness must be decided by an atomic call to Empty() in caller, due to the
```

Reply:
Also, what do you call "atomic" here? I don't see any synchronization around calls to `Empty`.

Reply:
"single call" might be a better description.

Reply:
Changed to "single call". Thanks for suggesting the new naming.